document/src/main/java/com/yahoo/document/idstring/IdString.java

```diff
@@ -43,14 +44,28 @@ public String getGroup() {
     private final String namespaceSpecific;
     private Utf8String cache;
     // This max unsigned 16 bit integer - 1 as the offset will be length + 1
-    static final int MAX_LENGTH_EXCEPT_NAMESPACE_SPECIFIC = 0xfffe;
+    static final int MAX_LENGTH_EXCEPT_NAMESPACE_SPECIFIC = 0xff00;
+    public static final int MAX_LENGTH = 0x10000;
 
     /**
      * Creates a IdString based on the given document id string.
      *
      * The document id string can only contain text characters.
      */
     public static IdString createIdString(String id) {
+        if (id.length() >= MAX_LENGTH) {
```

Comment:
Should this be `>` rather than `>=`? `MAX_LENGTH` sounds like it should be possible to create an ID of up to and including the given length

Reply:
Correct fixed